backend/src/main/kotlin/hu/bme/sch/cmsch/model/UserDetailsByInternalIdMappingEntity.kt

```diff
@@ -0,0 +1,100 @@
+
+    @JsonView(value = [ Edit::class ])
+    @Column(nullable = false)
+    @property:GenerateInput(order = 1, label = "PéK internal id",
+        note = "Ez módosítható eseti hiba kezelésre", enabled = true)
+    @property:GenerateOverview(visible = false)
+    @property:ImportFormat(ignore = false, columnId = 0)
+    var internalId: String? = null,
+
+    @JsonView(value = [ Edit::class, FullDetails::class ])
+    @property:GenerateInput(order = 2, label = "Neptun kód", enabled = true,
+        note = "Ez módosítható eseti hiba kezelésre", maxLength = 6)
+    @property:GenerateOverview(columnName = "Neptun", order = 2)
+    @property:ImportFormat(ignore = false, columnId = 1)
+    var neptun: String? = null,
+
+    @JsonView(value = [ Edit::class, FullDetails::class ])
+    @Enumerated(EnumType.STRING)
+    @property:GenerateInput(type = INPUT_TYPE_BLOCK_SELECT, order = 7, label = "Jogkör",
+        source = [ "GUEST", "BASIC", "ATTENDEE", "PRIVILEGED", "STAFF", "ADMIN", "SUPERUSER" ], minimumRole = RoleType.ADMIN,
+        note = "BASIC = belépett, STAFF = rendező, ADMIN = minden jog")
+    @property:GenerateOverview(visible = false)
+    @property:ImportFormat(ignore = false, columnId = 6, enumSource = RoleType::class, defaultValue = "GUEST")
+    var role: RoleType? = null,
+
+    @JsonView(value = [ Edit::class, Preview::class, FullDetails::class ])
+    @Column(nullable = false)
+    @property:GenerateInput(type = INPUT_TYPE_ENTITY_SELECT, order = 8, label = "Csoport", entitySource = "GroupEntity", minimumRole = RoleType.STAFF)
+    @property:GenerateOverview(columnName = "Csoport", centered = true, order = 3)
+    var groupName: String? = null,
+
+    @JsonIgnore
+    @ManyToOne(targetEntity = GroupEntity::class, fetch = FetchType.EAGER)
+    var group: GroupEntity? = null,
+
+    @JsonView(value = [ Edit::class, Preview::class, FullDetails::class ])
+    @Enumerated(EnumType.STRING)
+    @property:GenerateInput(type = INPUT_TYPE_BLOCK_SELECT, order = 9, label = "Gárda", source = [ "UNKNOWN", "BLACK", "BLUE", "RED", "WHITE", "YELLOW" ])
+    @property:GenerateOverview(columnName = "Gárda", centered = true, order = 5)
+    @property:ImportFormat(ignore = false, columnId = 7, type = IMPORT_ENUM, enumSource = GuildType::class, defaultValue = "UNKNOWN")
+    var guild: GuildType? = null,
+
+    @JsonView(value = [ Edit::class, Preview::class, FullDetails::class ])
+    @Enumerated(EnumType.STRING)
+    @property:GenerateInput(type = INPUT_TYPE_BLOCK_SELECT, order = 10, label = "Szak", source = [ "UNKNOWN", "IT", "EE", "BPROF" ])
+    @property:GenerateOverview(visible = false)
+    @property:ImportFormat(ignore = false, columnId = 8, type = IMPORT_ENUM, enumSource = MajorType::class, defaultValue = "UNKNOWN")
+    var major: MajorType? = null,
+
+    @Lob
+    @JsonView(value = [ Edit::class ])
+    @Column(nullable = false)
+    @property:GenerateInput(order = 16, label = "Jogosultságok", enabled = true, type = INPUT_TYPE_PERMISSIONS)
```

Comment:
Orders should be incremental. The UI will generate the input fields in this order.

Reply:
What is the difference between the generateInput and generateOverview annotations? Both have to be incremental or only the input?

Reply:
- property:GenerateOwerview will generate a table out of the entities:
![image](https://user-images.githubusercontent.com/1366778/215541459-dc6d96f9-b270-4eac-9658-c99bb38256d1.png)

- property:GenerateInput will generate UI to create or edit a given entity
![image](https://user-images.githubusercontent.com/1366778/215541565-2954a8e6-c3da-459c-aa65-11940bc47f53.png)

- property:ImportFormat will be used for importing and exporting to csv. 

- For the forst two, the order will determine its place in the UI. Having the same order id is not recommended. For a new entity it is recommended to be incremental.

- For the input format the first field that is not ignored must be 0. The order of the fields is not important, but no gaps allowed in the indexes. For example: If you have a 6 somewhere, you must have 0, 1, 2, 3, 4 and 5 as well.